catalog/core/catalog-core-api/src/main/java/ddf/catalog/CatalogFramework.java

```diff
@@ -117,6 +117,42 @@ CreateResponse create(CreateStorageRequest createRequest)
             throws IngestException, SourceUnavailableException;
 
     /**
+     * Creates {@link Metacard}s in the {@link ddf.catalog.source.CatalogProvider} and overrides valid attributes
+     * from the attributeOverrides map on each metacard.
+     *
+     * <p>
+     * <b>Implementations of this method must:</b>
+     * <ol>
+     * <li/>Before evaluation, call {@link ddf.catalog.plugin.PreIngestPlugin#process(CreateRequest)} for each
+     * registered {@link ddf.catalog.plugin.PreIngestPlugin} in order determined by the OSGi SERVICE_RANKING
+     * (Descending, highest first), "daisy chaining" their responses to each other.
+     * <li/>Call {@link ddf.catalog.source.CatalogProvider#create(CreateRequest)} on the registered
+     * {@link ddf.catalog.source.CatalogProvider}
+     * <li/>Call {@link ddf.catalog.plugin.PostIngestPlugin#process(CreateResponse)} for each registered
+     * {@link ddf.catalog.plugin.PostIngestPlugin} in order determined by the OSGi SERVICE_RANKING (Descending, highest
+     * first), "daisy chaining" their responses to each other.
+     * <li/>Call {@link ddf.catalog.content.plugin.PreCreateStoragePlugin#process(CreateStorageRequest)} for each
+     * registered {@link ddf.catalog.content.plugin.PreCreateStoragePlugin} in order determined by the OSGi SERVICE_RANKING
+     * (Descending, highest first), "daisy chaining" their responses to each other.
+     * <li/>Call {@link ddf.catalog.content.StorageProvider#create(CreateStorageRequest)} on the registered
+     * {@link ddf.catalog.source.CatalogProvider}
+     * <li/>Call {@link ddf.catalog.content.plugin.PostCreateStoragePlugin#process(ddf.catalog.content.operation.CreateStorageResponse)} for each registered
+     * {@link ddf.catalog.content.plugin.PostCreateStoragePlugin} in order determined by the OSGi SERVICE_RANKING (Descending, highest
+     * first), "daisy chaining" their responses to each other.
+     * </ol>
+     * </p>
+     * @param createRequest
+     *            the {@link CreateRequest}
+     * @return {@link CreateResponse}
+     * @throws IngestException
+     *             if an issue occurs during the update
+     * @throws SourceUnavailableException
+     *             if the source being updated is unavailable
+     */
+    CreateResponse create(CreateStorageRequest createRequest, Map<String, String> attributeOverrides)
```

Comment:
The attributeOverrides should be put in the create request properties.  This does not require a new method on catalog framework.